### PR TITLE
Injected query into location

### PR DESCRIPTION
**Pull request by codejockie**
This pull request adds the `query` object to `location`. This is useful when a user needs to access the value(s) in the search/query string without having to parse the search string themself.

### CODE REVIEW

File: src/reducer.js [hunk 1]
```diff
@@ -1,5 +1,32 @@
 import { LOCATION_CHANGE } from './actions'
 
+/**
+ * Adds query to location.
+ * Utilises the search prop of location to construct query.
+ */
+const injectQuery = (location) => {
+  if (!location) return location
```
Comment (supasate) [inline src/reducer.js]: if (!location) {
  return location
}

File: src/reducer.js [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+  const searchQuery = location.search || window.location.search
+
+  if (typeof searchQuery !== 'string' || searchQuery.length === 0) {
+    return { ...location, query: {} }
+  }
+
+  // Ignore the `?` part of the search string e.g. ?username=codejockie
+  const search = searchQuery.substring(1)
+
```
Comment (supasate) [inline src/reducer.js]: remove blank line

File: src/reducer.js [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+  // Split the query string on `&` e.g. ?username=codejockie&name=Kennedy
+  const queries = search.split('&')
+  // Contruct query
+  const query = queries.reduce((acc, currentQuery) => {
+    // Split on `=`, to get key and value
+    const [queryKey, queryValue] = currentQuery.split('=')
+    return { ...acc, [queryKey]: queryValue }
```
Comment (supasate) [inline src/reducer.js]: `
return {
  ...acc,
  [queryKey]: queryValue,
}
`

File: src/reducer.js [hunk 4]
```diff
@@ -1,5 +1,32 @@
...
+  }, {})
+
+  return { ...location, query }
```
Comment (supasate) [inline src/reducer.js]:
```
return {
   ...location,
   query,
}
```

**Comment by codejockie** [comment]
@supasate I have fixed the code style.

**Comment by supasate** [comment]
@codejockie Thanks so much for your contribution!

**Comment by storrella** [comment]
@supasate do you plan to build a release including these changes?

**Comment by sethomas** [comment]
https://github.com/supasate/connected-react-router/issues/376

**Comment by KorobkaVadim** [comment]
I use it with ElectronJS. Thanks for the lovely Monday morning with crash my application.
```
ReferenceError: window is not defined
    at injectQuery
```

**Comment by KorobkaVadim** [comment]
I can fix this error and create PR if no one objects and does not do it now? @codejockie @supasate

**Comment by codejockie** [comment]
> I can fix this error and create PR if no one objects and does not do it now? @codejockie @supasate

I was thinking of doing that. Since you are already open to do it, you can go ahead.

**Comment by KorobkaVadim** [comment]
https://github.com/supasate/connected-react-router/pull/377

**Comment by sixertoy** [comment]
Hello :)
Seems like it was not a good need in my case.
I've got to manage URL from a third party app, and this URL contains a parameter called `query` so... voilà voilà :)
May be could we have an option to disable this query parameter ?

**Pull request merged**